PIL/SgiImagePlugin.py

```diff
@@ -123,28 +177,59 @@ def _save(im, fp, filename):
     fp.write(struct.pack('>H', z))
     fp.write(struct.pack('>l', pinmin))
     fp.write(struct.pack('>l', pinmax))
-
     fp.write(struct.pack('4s', b''))  # dummy
     fp.write(struct.pack('79s', imgName))  # truncates to 79 chars
     fp.write(struct.pack('s', b''))  # force null byte after imgname
     fp.write(struct.pack('>l', colormap))
-
     fp.write(struct.pack('404s', b''))  # dummy
 
-    # assert we've got the right number of bands.
-    if len(im.getbands()) != z:
-        raise ValueError("incorrect number of bands in SGI write: %s vs %s" %
-                         (z, len(im.getbands())))
-
     for channel in im.split():
-        fp.write(channel.tobytes())
+        rawchannel = channel.tobytes()
+        if bpc == 1:
+            fp.write(rawchannel)
+        else:
+            for pixel in rawchannel:
+                fp.write(o16(i8(pixel) * 256))
```

Comment:
Should probably do this in C by converting to `I`, and using  ImageMath to multiply by 256.

(though, somewhere there should be a MSB promotion to `I` or `I;16`, as that's a common issue)